addons/mail/static/src/core/common/message_model.js

```diff
@@ -362,6 +362,51 @@ export class Message extends Record {
         );
     }
 
+    /**
+     * Checks if the message requires a preview.
+     * A message needs a preview if its body is empty and it has one or more attachments.
+     */
+    get requiresPreview() {
+        return this.isBodyEmpty && this.attachment_ids.length > 0;
+    }
+
+    get messagePreviewText() {
+        if (!this.requiresPreview) {
+            return this.inlineBody || this.subtype_description;
+        }
+        const { attachment_ids: attachments } = this;
+        if (!attachments || attachments.length === 0) return "";
+
+        switch (attachments.length) {
+            case 1:
+                return attachments[0].previewName;
+            case 2:
+                return _t("%(firstAttachmentName)s and %(secondAttachmentName)s", {
+                    firstAttachmentName: attachments[0].previewName,
+                    secondAttachmentName: attachments[1].previewName,
+                    count: attachments.length - 1,
+                });
+            default:
+                return _t("%(firstAttachmentName)s and %(count)s other attachments", {
+                    firstAttachmentName: attachments[0].previewName,
+                    count: attachments.length - 1,
+                });
```

Comment:
It's easier for translator to format translation string that way:
```
"%(file1)s and %(file2)s"
"%(file1)s and %(count)s other attachments"
```
The context is good enough with `file1` rather than `firstAttachmentName`. Readability is best with shortest word that is specific enough for the context.